geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/rebalance/model/PartitionedRegionLoadModel.java

```diff
@@ -425,19 +425,58 @@ public void remoteOverRedundancyBucket(BucketRollup bucket, Member targetMember)
   }
 
   private void initLowRedundancyBuckets() {
-    this.lowRedundancyBuckets = new TreeSet<>(REDUNDANCY_COMPARATOR);
     for (BucketRollup b : this.buckets) {
       if (b != null && b.getRedundancy() >= 0 && b.getRedundancy() < this.requiredRedundancy) {
         this.lowRedundancyBuckets.add(b);
       }
     }
   }
 
+
+  /**
+   * Original functionality if bucket's redundancy is greater than what is necessary add it to the
+   * over redundancy bucket list, so it can be cleared
+   * <p>
+   * Newly added functionality is to make this so that we don't have a bucket in the same redundancy
+   * zone twice if zones are in use.
+   */
   private void initOverRedundancyBuckets() {
     this.overRedundancyBuckets = new TreeSet<>(REDUNDANCY_COMPARATOR);
+    Set<String> redundancyZonesFound = new HashSet<>();
+
+    // For every bucket
     for (BucketRollup b : this.buckets) {
-      if (b != null && b.getOnlineRedundancy() > this.requiredRedundancy) {
-        this.overRedundancyBuckets.add(b);
+      if (b != null) {
+        // check to see if the existing redundancy is greater than required
+        if (b.getOnlineRedundancy() > this.requiredRedundancy) {
+          // if so, add the bucket to the over redundancy list
+          this.overRedundancyBuckets.add(b);
+        } else {
```

Comment:
I think it would be good move the logic in the else block to a separate method for testability/readability. This whole stanza is really just asking "are two copies in the same zone".

Reply:
Yup we talked about that. I will make that change.